dashboard/test/factories/factories.rb

```diff
@@ -240,17 +231,35 @@
         end
       end
 
-      factory :google_oauth2_student do
-        encrypted_password nil
-        provider 'google_oauth2'
-        sequence(:uid) {|n| n}
-      end
```

Comment:
I switched from factories to traits for these SSO users, because the configuration ends up being exactly the same for teachers and students.

This factory was also used in `UserTest` - I updated those places.  Heads-up that this might break tests in other work-in-progress PRs.